xtask/src/print_snapshot.rs

```diff
@@ -0,0 +1,73 @@
+use crate::do_test;
+use crate::snapshot::{all_snapshot_tests, Snapshot};
+use crate::utils::{load_expected_output, run_capturing_stdout};
+use anyhow::{anyhow, Context};
+use colored::Colorize;
+use similar::{ChangeTag, TextDiff};
+use std::process::{Command, Stdio};
+
+pub fn test_print_snapshot(snapshot: Option<Snapshot>) {
+    match snapshot {
+        None => test_all_print_snapshots(),
+        Some(snapshot) => {
+            do_test(
+                || test_single_print_snapshot(snapshot.name()),
+                "qemu/snapshot_print",
+            );
+        }
+    }
+}
+
+pub fn test_all_print_snapshots() {
+    for test in all_snapshot_tests() {
+        do_test(|| test_single_print_snapshot(test), "qemu/snapshot_print");
+    }
+}
+
+pub fn test_single_print_snapshot(name: &str) -> anyhow::Result<()> {
+    println!("{}", name.bold());
+
+    let frame_path = format!("xtask/output_files/{}.out", name);
+    let elf_path = format!("xtask/snapshot_elfs/{}", name);
+
+    let frames = Command::new("cat")
+        .arg(frame_path)
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let actual = run_capturing_stdout(
+        Command::new("defmt-print")
+            .arg("-e")
+            .arg(elf_path)
+            .arg("--log-format")
+            .arg("{L:4} {s}")
+            .stdin(Stdio::from(frames.stdout.unwrap())),
+    )
+    .with_context(|| name.to_string())?;
```

Comment:
@jonathanpallant Clippy refers to the `Command` in line 33.

I think it is correct without `.wait`, because it pipes the output from `cat` into `defmt-print`?. So we do _not_ want to wait until cat exited before executing `run_capturing_stdout`.

Iiuc the pipe also makes sure cat does not exit before we execute `run_capturing_stdout`.

So we just need to allow it.
```suggestion
    #[allow(
        clippy::zombie_processes,
        reason = "no wait needed, because we pipe into defmt-print"
    )]
    let frames = Command::new("cat")
        .arg(frame_path)
        .stdout(Stdio::piped())
        .spawn()
        .unwrap();

    let actual = run_capturing_stdout(
        Command::new("defmt-print")
            .arg("-e")
            .arg(elf_path)
            .arg("--log-format")
            .arg("{L:4} {s}")
            .stdin(Stdio::from(frames.stdout.unwrap())),
    )
    .with_context(|| name.to_string())?;
```

Reply:
I think you could also bypass the issue by not spawning `cat` and instead doing `std::fs::read()` on the file and passing the bytes to the stdin of the `defmt-print` process.

Reply:
True, that sounds simpler actually.